src/log.ts

```diff
@@ -23,28 +34,71 @@ export interface ILogger {
  * log.on(log.ERROR, console.error.bind(console));
  * ```
  */
-export class Log extends MicroEmitter implements ILogger {
+export class Log extends MicroEmitter<EmittedEvents> {
     /**
      * The Debug event constant.
      */
-    readonly DEBUG = 'debug';
+    readonly DEBUG = DEBUG;
     /**
      * The info event constant.
      */
-    readonly INFO = 'info';
+    readonly INFO = INFO;
     /**
      * The warn event constant.
      */
-    readonly WARN = 'warn';
+    readonly WARN = WARN;
     /**
      * the error event constant.
      */
-    readonly ERROR = 'error';
+    readonly ERROR = ERROR;
+
+    constructor() {
+        super();
+        this.error = this.error.bind(this);

```

Comment:
Changed arrow function to binding for better dev experience since IDE is able to better show arg hints and docs if a class has a method instead of a property.